File: app/src/main/java/me/echeung/moemoekyun/api/endpoints/Kpop.java
package me.echeung.moemoekyun.api.endpoints;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Kpop extends Library {

    public static final Library INSTANCE = new Kpop();

    public static final String NAME = "kpop";

    public String getName() {
        return NAME;
    }

    public String getSocketUrl() {
        return "wss://listen.moe/kpop/gateway";
    }

    public String getStreamUrl() {
        return "https://listen.moe/kpop/fallback";
    }

}

File: app/src/main/java/me/echeung/moemoekyun/client/api/v5/library/Kpop.kt
package me.echeung.moemoekyun.client.api.v5.library

class Kpop private constructor() : Library(
        NAME,
        "wss://listen.moe/kpop/gateway",
        "https://listen.moe/kpop/fallback") {
    companion object {
        const val NAME = "kpop"

        val INSTANCE: Library = Kpop()
    }
}
